component/src/main/java/org/wso2/rule/validator/document/Document.java

```diff
@@ -290,4 +307,99 @@ public static List<String> toPath(String path) {
 
         return segments;
     }
+
+    public Object getRootDocument() {
```

Comment:
Let's add Java doc comments for all the public methods, and also have them for private methods is also recommended as it will help out developers